pkg/cvo/configuration/configuration_test.go

```diff
@@ -0,0 +1,276 @@
+			}
+
+			// Shutdown created resources
+			cancelFunc()
+		})
+	}
+}
+
+func TestClusterVersionOperatorConfiguration_Sync(t *testing.T) {
+	tests := []struct {
+		name           string
+		config         *operatorv1alpha1.ClusterVersionOperator
+		expectedConfig *operatorv1alpha1.ClusterVersionOperator
+	}{
+		{
+			name:           "the configuration resource does not exist in the cluster -> ignore",
+			config:         nil,
+			expectedConfig: nil,
+		},
+		{
+			name: "Sync updates the ClusterVersionOperator resource",
+			config: &operatorv1alpha1.ClusterVersionOperator{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:       "cluster",
+					Generation: 4,
+				},
+				Spec: operatorv1alpha1.ClusterVersionOperatorSpec{
+					OperatorLogLevel: operatorv1.Trace,
+				},
+				Status: operatorv1alpha1.ClusterVersionOperatorStatus{
+					ObservedGeneration: 3,
+				},
+			},
+			expectedConfig: &operatorv1alpha1.ClusterVersionOperator{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:       "cluster",
+					Generation: 4,
+				},
+				Spec: operatorv1alpha1.ClusterVersionOperatorSpec{
+					OperatorLogLevel: operatorv1.Trace,
+				},
+				Status: operatorv1alpha1.ClusterVersionOperatorStatus{
+					ObservedGeneration: 4,
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Initialize testing logic
+			var client *operatorclientsetfake.Clientset
+			if tt.config != nil {
+				client = operatorclientsetfake.NewClientset(tt.config)
+			} else {
+				client = operatorclientsetfake.NewClientset()
+			}
+
+			ctx, cancelFunc := context.WithDeadline(context.Background(), time.Now().Add(time.Minute))
+			factory := operatorexternalversions.NewSharedInformerFactoryWithOptions(client, time.Minute)
+			cvoConfiguration := NewClusterVersionOperatorConfiguration(client, factory)
+			err := cvoConfiguration.Start(ctx)
+			if err != nil {
+				t.Errorf("unexpected error %v", err)
+			}
+
+			// Run tested functionality
+			err = cvoConfiguration.Sync(ctx, "ClusterVersionOperator/cluster")
+			if err != nil {
+				t.Errorf("unexpected error %v", err)
+			}
+
+			// Verify results
+			config, err := client.OperatorV1alpha1().ClusterVersionOperators().Get(ctx, "cluster", metav1.GetOptions{})
+			if err != nil && !apierrors.IsNotFound(err) {
+				t.Errorf("unexpected error %v", err)
+			}
+
+			switch {
+			case apierrors.IsNotFound(err) && tt.expectedConfig == nil:
+			case apierrors.IsNotFound(err) && tt.expectedConfig != nil:
+				t.Errorf("expected config to be '%v', got NotFound", *tt.expectedConfig)
+			case config != nil && tt.expectedConfig == nil:
+				t.Errorf("expected config to be NotFound, got '%v'", *config)
+			case config != nil && tt.expectedConfig != nil:
+				if diff := cmp.Diff(*tt.expectedConfig, *config, cmpopts.IgnoreFields(operatorv1alpha1.ClusterVersionOperator{}, "ObjectMeta")); diff != "" {
+					t.Errorf("unexpected config (-want, +got) = %v", diff)
+				}
```

Comment:
Just chose any part that you like:

```suggestion
			switch {
			case apierrors.IsNotFound(err) && tt.expectedConfig != nil:
				t.Errorf("expected config to be '%v', got NotFound", *tt.expectedConfig)
			case err == nil:
				if diff := cmp.Diff(tt.expectedConfig, config, cmpopts.IgnoreFields(metav1.ObjectMeta{}, "ManagedFields")); diff != "" {
					t.Errorf("unexpected config (-want, +got) = %v", diff)
				}
			}
```

Reply:
Thanks!

Done in https://github.com/openshift/cluster-version-operator/pull/1170/commits/ba839395ead1930eeb16bbffb2d9e1b574833380